### PR TITLE
Experimental: Use Hooks

### PR DESCRIPTION
**Pull request by dt-ap**
Convert four components from using **Class Component** to **Function Component** using **Hooks**.

### CODE REVIEW

File: modern/src/DeviceList.js [hunk 1]
```diff
@@ -1,96 +1,79 @@
...
-          <MenuItem onClick={() => this.handleMenuRemove()}>{t('sharedRemove')}</MenuItem>
-        </Menu>
-        <RemoveDialog
-          open={this.state.removeDialogOpen}
-          onClose={() => { this.setState({ removeDialogOpen: false }) }} />
-      </Fragment>
-    );
-  }
+  return (
+    <>
```
Comment (tananaev) [inline modern/src/DeviceList.js]: Is this fragment?

File: modern/src/MainToolbar.js [hunk 2]
```diff
@@ -32,149 +33,124 @@ const styles = theme => ({
...
-                </ListItemIcon>
-                <ListItemText primary={t('sharedNotifications')} />
-              </ListItem>
-            </List>
-          </div>
-        </Drawer>
-      </Fragment>
-    );
-  }
+  return (<>
```
Comment (tananaev) [inline modern/src/MainToolbar.js]: Please move tag to the next line

File: modern/src/MainToolbar.js [hunk 3]
```diff
@@ -32,149 +33,124 @@ const styles = theme => ({
...
+          <ListItem button disabled>
+            <ListItemIcon>
+              <SettingsIcon />
+            </ListItemIcon>
+            <ListItemText primary={t('sharedNotifications')} />
+          </ListItem>
+        </List>
+      </div>
+    </Drawer>
+  </>);
```
Comment (tananaev) [inline modern/src/MainToolbar.js]: Same with closing

File: modern/src/DeviceList.js (hunk 1 above)
Comment (dt-ap) [reply modern/src/DeviceList.js]: Yes

File: modern/src/MainPage.js [hunk 4]
```diff
@@ -35,57 +36,45 @@ const styles = theme => ({
...
-              </ContainerDimensions>
-            </div>
-          </div>
+  return loading ? (<div>Loading...</div>) : (
+    <div className={classes.root}>
+      <SocketController />
+      <MainToobar />
+      <div className={classes.content}>
+        {/* <Drawer
+          anchor={isWidthUp('sm', this.props.width) ? "left" : "bottom"} NOTE: What's this do?
```
Comment (tananaev) [inline modern/src/MainPage.js]: There is a desktop and mobile version. Please keep the original code. Are there any other changes you've made?

Comment (dt-ap) [reply modern/src/MainPage.js]: I think there are not. But without using `isWidthUp` the permanent drawer moves to bottom automatically

Comment (tananaev) [reply modern/src/MainPage.js]: Let's revert this change. If you want to change something, I'm happy to review and discuss it, but let's not mix migration and changes together. It makes it impossible to review.

File: modern/src/MainPage.js [hunk 5]
```diff
@@ -35,57 +37,42 @@ const styles = theme => ({
...
-              </ContainerDimensions>
-            </div>
-          </div>
+  return loading ? (<div>Loading...</div>) : (
+    <div className={classes.root}>
+      <SocketController />
+      <MainToobar />
+      <div className={classes.content}>
+        <Drawer
+          anchor={isWidthUp('sm', width) ? "left" : "bottom"}
```
Comment (dt-ap) [inline modern/src/MainPage.js]: @tananaev Material-UI use different API if you want the hooks alternative. Here the [docs](https://material-ui.com/components/use-media-query/)

**Comment by tananaev** [comment]
Merged, thanks.

**Pull request merged**